### ISSUE TITLE
Friendly chat tutorial disappear in Google code lab?

### ISSUE THREAD

**Issue by CircleCurve** [flutter/flutter#17380]
I look at the google web cache ...and it has changed to another tutorial (baby vote) ?
[Current tutorial link](https://codelabs.developers.google.com/codelabs/flutter-firebase/#1)

[Google web cache for friendly chat](https://webcache.googleusercontent.com/search?q=cache:ugkJTlrKoaIJ:https://codelabs.developers.google.com/codelabs/flutter-firebase/+&cd=1&hl=zh-TW&ct=clnk&gl=hk#2)

![image](https://user-images.githubusercontent.com/8490834/39739154-034e84a8-52c2-11e8-9ec4-1c9d886a995a.png)

Seems like it changes to baby name vote (?
![image](https://user-images.githubusercontent.com/8490834/39739175-1f7e617a-52c2-11e8-9bc5-f6d5f84d62d4.png)

I still hope that can maintain chat app tutorial. It' s really good way for step by step to understand how to make a whole chat app from row, column, layout, animation, button interactive and firebase.  Not just the part of it

**Comment by MisterJimson** [comment]
This one? https://codelabs.developers.google.com/codelabs/flutter/#0

**Comment by allandaleho** [comment]
Yes was doing the tutorial last night when they suddenly changed it.. I shouldn't have refreshed the page :(

Is there any way I can continue with the tutorial? I really want to finish that one atleast.

**Comment by CircleCurve** [comment]
I also use web cache and webarchive to find the cache of friendly chat tutorial but no luck . Can anybody help us :(

**Comment by mrmax99** [comment]
I was also in the middle of this tutorial yesterday and today it is gone.  I was running into infuriating dependency issues at step 5 (the old one, google sign-in step), so I hope that at least the replacement tutorial isn't completely broken.  I wish the tutorial was just fixed or a note was there it was going to be deleted soon so I didn't waste my time.

**Comment by CircleCurve** [comment]
@rohan20 Yes, bro your git repo is really well but we still need to dig into the code to understand what is going on here. I hope that there is a comprehensive with picture and step by step tutorial. It must be the easiest way to follow it. Rather than know how, i want to know why. Thanks bro

**Comment by allandaleho** [comment]
@mrmax99 was having some issues with google sign-in also, found a solution last night. something about force update/upgrade the packages (would take about 3 mins) and everything worked out.. I've also updated the package version before doing the upgrade. 

I will attach here the solution for those who got stuck with the same, but really hoping that they bring the tutorial back up as the replacement does not seem to include much compared to the last one(such as analytics, sign in, etc..). 

Anyone has a copy of it so we can continue with the tutorial? I have the copy of the git step by step, but I really like the actual tutorial so I can fully grasp the concept behind the codes. :(

**Comment by CircleCurve** [comment]
@allandaleho totally agree with you bro. Still waiting someone can help newbie for flutter :(

**Comment by allandaleho** [comment]
The fix I followed for fixing my dependency issues were to change my pubspec.yaml dependecies to  
`
  image_picker: ^0.3.0
  google_sign_in: ^3.0.0
  firebase_analytics: ^0.3.0
  firebase_auth: ^0.5.0
  firebase_database: ^0.4.0
  firebase_storage: ^0.2.0 
`

and do a 
`flutter update-packages --force-upgrade`

About the tutorial missing, should we just wait for response from flutter team or should we mention them here?

**Comment by CircleCurve** [comment]
I think we should mention them here. Friendly  chat tutorial cover different important topic of flutter from UI design to firebase backend. And also, friendly chat is the one of flutter apps, show in google io or ..some exhibition? To be honest, I study flutter because of the first impression of chat app.  As a flutter's newbie, it is very useful for understand the concept.

**Comment by CircleCurve** [comment]
Sorry about that, which one we should mention or tag for ? @allandaleho

**Comment by allandaleho** [comment]
@CircleCurve just to update I was able to finish the tutorial using firebase and now trying it in firestore.

**Comment by wmleler** [comment]
Hi. I wrote the new firebase-flutter codelab. It wasn't my decision to deprecate the old friendly chat version, but there were a number of problems with it. It was way too long and most people were not finishing it. So we are trying to divide the Firebase codelab up into smaller pieces that are each specific to a feature of Firebase. The old codelab also needed significant updating.

The old code is still around in a repository (I added a link to that repository to the new codelab), so feel free to use that if you want an example of a chat app using Firebase and Flutter. And I encourage you to update it as you discover things that are out of date.

It was my decision to replace the old codelab with the new one. We did that because for a while the old codelab was the only public document with instructions on how to connect a Flutter app to Firebase (even though its instructions were out of date -- with old versions of libraries, and several steps that are no longer necessary). But unfortunately there are a number of articles out there that include links to the codelab just to tell people how to connect, and we didn't want to break those links.

To all the people who were in the middle of the codelab when we did the switch, I'm sorry. There will be more codelabs coming for Flutter and Firebase.
